### PR TITLE
Update build workflow bicep

### PR DESCRIPTION
**Pull request by sk593**

### CODE REVIEW

File: .github/scripts/get_release_version.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+gitRef = os.getenv("GITHUB_REF")
+
+# From https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+# Group 'version' returns the whole version
+# other named groups return the components
+tagRefRegex = r"^refs/tags/v(?P<version>0|(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$"
+pullRefRegex = r"^refs/pull/(.*)/(.*)$"
+
+with open(os.getenv("GITHUB_ENV"), "a") as githubEnv:
```
Comment (lakshmimsft) [inline .github/scripts/get_release_version.py:63]: should we handle if "GITHUB_ENV" is not explicitly set, it exits cleanly

Comment (sk593) [reply .github/scripts/get_release_version.py:63]: I pulled this file from the radius repo so it might be out of scope of this PR to change that (i.e. if we change one, we should change all)

File: .github/workflows/publish-bicep.yaml [hunk 2]
```diff
@@ -16,13 +16,77 @@
...
+        if: github.repository == 'radius-project/bicep-types-aws' && ((startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'))
+        run: |
+          curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64
+          chmod +x ./bicep
+          bicep --version
+      - name: Publish bicep types 
+        if: github.repository == 'radius-project/bicep-types-aws' && ((startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'))
+        run: |
+          bicep publish-provider ./artifacts/bicep/index.json --target br:${{ env.BICEP_TYPES_REGISTRY }}/aws:${{ env.REL_VERSION == 'edge' && 'latest' || env.REL_VERSION }} --force
   update-types:
```
Comment (lakshmimsft) [inline .github/workflows/publish-bicep.yaml:90]: not part of this PR but i'm seeing that 'set-output' on line 103, 104 is deprecated. 
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
maybe we can update it in this or another PR

Comment (sk593) [reply .github/workflows/publish-bicep.yaml:90]: Good catch, this will eventually be deleted but I updated it for now so the workflow runs fine

File: .github/scripts/get_release_version.py (hunk 1 above)
Comment (lakshmimsft) [reply .github/scripts/get_release_version.py:63]: yup, looks like it's typically something the github actions sets, not something we set. we should be okay

**Review by lakshmimsft — APPROVED** [review]

**Comment by lakshmimsft** [comment]
pls check all checks go through. thanks!

**Comment by sk593** [comment]
> pls check all checks go through. thanks!

Tested this through a PR to main first and the tests passed. Will also be tested as part of this larger PR: https://github.com/radius-project/bicep-types-aws/pull/43

File: .github/workflows/publish-bicep.yaml [hunk 3]
```diff
@@ -16,14 +16,79 @@
...
+    if: github.event_name == 'pull_request' && github.base_ref == 'main'
```
Comment (sk593) [inline .github/workflows/publish-bicep.yaml:91]: Updated so this only runs on pulls from main. We don't want this running on pushes
